R/composite_ppi.R

```diff
@@ -80,19 +88,25 @@ composite_ppi <- function(x, param = "DBZH", nx = 100, ny = 100, xlim, ylim, res
   if (!missing(res)) {
     assert_that(is.numeric(res))
     assert_that(length(res) <= 2)
+    t_res <- res
   } else {
-    res <- NA
+    t_res <- NULL
   }
-
   # check crs argument as in raster::raster()
   if (!missing(crs)) {
-    crs <- CRS(as.character(raster::projection(crs)))
-  }
-  else {
-    crs <- CRS("+proj=longlat +datum=WGS84")
+    t_crs <- CRS(as.character(raster::projection(crs)))
+  } else {
+    t_crs <- NULL
   }
+  if (!all(method %in% c("max", "min", "mean", "idw"))) stop("'method' should be one or multiple of 'max', 'mean', 'min' or 'idw'")
+  if (length(method) != length(param) & length(method) != 1) stop("'method' should be of length 1 or length(param)")
+  if (!is.logical(coverage)) stop("'coverage' should be a logical")
```

Comment:
@barthoekstra : A quick hint this can easier be written as `assert_that(is.flag(coverage))`. It directly ensures the length is 1

Reply:
fixed this in b89d6c37c7f96c4f05ee4d8588328a3f70b0612b